packages/mobile/src/identity/privacy.ts

```diff
@@ -24,43 +30,56 @@ export interface PhoneNumberHashDetails {
 // Fetch and cache a phone number's salt and hash
 export function* fetchPhoneHashPrivate(e164Number: string) {
   try {
-    Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Fetching phone hash details')
-    const saltCache: E164NumberToSaltType = yield select(e164NumberToSaltSelector)
-    const cachedSalt = saltCache[e164Number]
-
-    if (cachedSalt) {
-      Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was cached')
-      const phoneHash = getPhoneHash(e164Number, cachedSalt)
-      return { e164Number, phoneHash, salt: cachedSalt }
-    }
-
-    Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was not cached, fetching')
-    const account: string = yield select(currentAccountSelector)
-    const contractKit: ContractKit = yield call(getContractKit)
-    const selfPhoneHash: string | undefined = yield call(getUserSelfPhoneHash)
-    const details: PhoneNumberHashDetails = yield call(
-      getPhoneHashPrivate,
-      e164Number,
-      account,
-      contractKit,
-      selfPhoneHash
-    )
-    yield put(updateE164PhoneNumberSalts({ [e164Number]: details.salt }))
+    const details: PhoneNumberHashDetails = yield call(doFetchPhoneHashPrivate, e164Number)
     return details
   } catch (error) {
     if (error.message === ErrorMessages.SALT_QUOTA_EXCEEDED) {
       Logger.error(
         `${TAG}@fetchPrivatePhoneHash`,
         'Salt quota exceeded, navigating to quota purchase screen'
       )
-      // TODO nav to quota purchase screen
+      const result: boolean = yield call(navigateToQuotaPurchaseScreen)
+      // If quota purchase was successful, try lookup a second time
+      if (result) {
```

Comment:
Nit: I would find this more readable if `result` were `quotaPurchaseSucess`

Reply:
Will do!